CMakeLists.txt

```diff
@@ -1,128 +1,142 @@
+endif()
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
+# Warnings & link hygiene
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
+  add_link_options(-Wl,-no-undefined)
+endif()
+
+# Enable ASAN in Debug builds
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  message(STATUS "Enabling AddressSanitizer (ASAN) for Debug build")
+  set(ASAN_FLAGS "-fsanitize=address -fno-omit-frame-pointer")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ASAN_FLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ASAN_FLAGS}")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_FLAGS}")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${ASAN_FLAGS}")
+endif()
+
+# Distro detection (override with -BUILD_HUMBLE=ON/OFF)
+if(NOT DEFINED BUILD_HUMBLE)
+  if(DEFINED ENV{ROS_DISTRO} AND "$ENV{ROS_DISTRO}" STREQUAL "humble")
+    set(BUILD_HUMBLE TRUE)
+    message(STATUS "ROS_DISTRO=humble -> using Catch2 v2")
+  else()
+    set(BUILD_HUMBLE FALSE)
+    message(STATUS "ROS_DISTRO>humble -> Catch2 v3 (Jazzy/Rolling)")
+  endif()
 endif()
 
-# find dependencies
+# Dependencies
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_lifecycle REQUIRED)
 find_package(rclcpp_components REQUIRED)
 find_package(can_msgs REQUIRED)
-# uncomment the following section in order to fill in
-# further dependencies manually.
-# find_package(<dependency> REQUIRED)
-
-set(dependencies
-  rclcpp
-  rclcpp_lifecycle
-  rclcpp_components
-  can_msgs
-)
 
+# Primary library
 add_library(socketcan_adapter SHARED
-            src/socketcan_adapter.cpp
-            src/socketcan_bridge_node.cpp
-            src/can_frame.cpp)
-
-target_compile_features(socketcan_adapter PUBLIC c_std_99 cxx_std_17)  # Require C99 and C++17
+  src/socketcan_adapter.cpp
+  src/socketcan_bridge_node.cpp
+  src/can_frame.cpp
+)
 target_include_directories(socketcan_adapter PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:include>)
-
-# Causes the visibility macros to use dllexport rather than dllimport,
-# which is appropriate when building the dll but not consuming it.
+  $<INSTALL_INTERFACE:include>
+)
 target_compile_definitions(socketcan_adapter PRIVATE "SOCKETCAN_ADAPTER_BUILDING_LIBRARY")
 
-ament_target_dependencies(socketcan_adapter ${dependencies})
+# Executable using the library
+add_executable(socketcan_bridge src/socketcan_bridge.cpp)
+target_include_directories(socketcan_bridge PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
 
-install(
-  DIRECTORY include/
-  DESTINATION include
+# Linkage: Humble keeps ament_target_dependencies; Jazzy/Rolling use imported targets
+target_link_libraries(socketcan_adapter PUBLIC
+  ${can_msgs_TARGETS}
+  rclcpp::rclcpp
+  rclcpp_components::component
+  rclcpp_components::component_manager
```

Comment:
Definitely don't need `component_manager`, and `component` is not currently used as far as I can tell here - though I would say you should go ahead and export the node as a component, because that's just good to do

```suggestion
```